File: platform/polyglot/rag_search_algorithms/mjs/src/context-builder.mjs
/**
 * @fileoverview Context builder for LLM prompts.
 */

/**
 * Build a formatted context string from search result documents.
 *
 * @param {Array<{metadata: Record<string, any>, pageContent?: string, page_content?: string}>} docs
 * @returns {string}
 */
export function buildContext(docs) {
  const contextParts = [];
  for (let i = 0; i < docs.length; i++) {
    const doc = docs[i];
    const meta = doc.metadata || {};
    const parts = [];
    if (meta.component) {
      parts.push(`component=${meta.component}`);
    }
    if (meta.file_name) {
      parts.push(meta.file_name);
    }
    const header = parts.join(' | ');
    const content = doc.pageContent || doc.page_content || '';
    contextParts.push(`--- Source ${i + 1}: ${header} ---\n${content}`);
  }
  return contextParts.join('\n\n');
}

File: platform/polyglot/rag_search_algorithms/py/rag_search_algorithms/context_builder.py
"""Context builder for LLM prompts."""


def build_context(docs: list) -> str:
    """Build a formatted context string from search result documents.

    Args:
        docs: List of document objects with .metadata and .page_content attributes.

    Returns:
        Formatted context string with source headers and content.
    """
    context_parts = []
    for i, doc in enumerate(docs, 1):
        parts = []
        meta = doc.metadata
        if "component" in meta:
            parts.append(f"component={meta['component']}")
        if "file_name" in meta:
            parts.append(meta["file_name"])
        header = " | ".join(parts)
        context_parts.append(f"--- Source {i}: {header} ---\n{doc.page_content}")
    return "\n\n".join(context_parts)
